lighthouse-core/gather/gather-runner.js

```diff
@@ -189,7 +190,29 @@ class GatherRunner {
     const driver = options.driver;
     const tracingData = {traces: {}};
 
-    passes = GatherRunner.instantiateGatherers(passes);
+    if (typeof options.url !== 'string' || options.url.length === 0) {
+      return Promise.reject(new Error('You must provide a url to the driver'));
+    }
+
+    if (typeof options.flags === 'undefined') {
+      options.flags = {};
+    }
+
+    if (typeof options.config === 'undefined') {
+      return Promise.reject(new Error('You must provide a config'));
+    }
+
+    // Default mobile emulation and page loading to true.
```

Comment:
This set of checks duplicates what's already in runner. : https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/runner.js#L29-L44

I think we should only check where neccessary and not end up with all the duplication. If we can't keep straight if we need a safety check, then we probably need to untangle a bit. (I'm thinking of our refactor for `traceEvents` a bit ago. There's a few more in the wings.)